src/Oscoin/Consensus/Nakamoto.hs

```diff
@@ -182,16 +187,18 @@ difficultyBlocks :: Depth
 difficultyBlocks = 2016
 
 -- | Calculate the difficulty of a blockchain.
-chainDifficulty :: [Block c tx s] -> Difficulty
+chainDifficulty :: [Block c tx s] -> Telemetry.Traced Difficulty
 chainDifficulty [] =
-    minDifficulty
+    pure minDifficulty
 chainDifficulty (NonEmpty.fromList -> blks)
-    | NonEmpty.length blks `mod` fromIntegral difficultyBlocks == 0 =
-        encodeDifficulty $ fst (decodeDifficulty currentDifficulty)
-                         * fromIntegral targetElapsed
-                         `div` toInteger actualElapsed
+    | NonEmpty.length blks `mod` fromIntegral difficultyBlocks == 0 = do
+        let newDifficulty =
+                encodeDifficulty $ fst (decodeDifficulty currentDifficulty)
+                                 * fromIntegral targetElapsed
+                                 `div` toInteger actualElapsed
+        Telemetry.traced (DifficultyAdjustedEvent newDifficulty currentDifficulty) newDifficulty
```

Comment:
I think this changes the meaning of the function, and should be moved out to a different function: `chainDifficulty` is meant to return the target chain difficulty - it isn't concerned about adjustment events, as it is meant to be used simply to calculate a value. Adding telemetry to it confuses its purpose. Furthermore, since difficulty is adjusted every 2016 blocks regardless of whether this function is called or not, the event should be fired every 2016 blocks from whichever place is receiving or producing blocks, not from a function which calculates difficulty of an input chain (which may not be the full chain). Thoughts? @kim @adinapoli-mndc

Reply:
> I think this changes the meaning of the function

Huh?

Reply:
Example: if I call this function to calculate the difficulty of a fork, I get a `DifficultyAdjustedEvent` - which would be wrong.

Reply:
> > I think this changes the meaning of the function
> 
> Huh?

This function now assumes a certain usage - which is that if the returned difficulty is a multiple of 2016, it should trigger an event that should be observable from our logs/metrics. This is only useful if used on every block and only on the main chain..

Reply:
We can argue about the meaning of the event, but I fail to see any wrongness nor alteration of the meaning of the function.